src/embind/embind.js

```diff
@@ -35,6 +35,29 @@ var LibraryEmbind = {
   $PureVirtualError__deps: ['$extendError'],
   $PureVirtualError: undefined,
   $GenericWireTypeSize: {{{ 2 * POINTER_SIZE }}},
+  $createEmValType__deps: ['_emval_decref', '$Emval', '$simpleReadValueFromPointer', '$GenericWireTypeSize'],
+  $createEmValType: () => {
+    return {
+      name: 'emscripten::val',
+      'fromWireType': (handle) => {
+        var rv = Emval.toValue(handle);
+        __emval_decref(handle);
+        return rv;
+      },
+      'toWireType': (destructors, value) => Emval.toHandle(value),
+      'argPackAdvance': GenericWireTypeSize,
+      'readValueFromPointer': simpleReadValueFromPointer,
+      destructorFunction: null, // This type does not need a destructor
+
+      // TODO: do we need a deleteObject here?  write a test where
+      // emval is passed into JS via an interface
+    };
+  },
+  // If register_type is used, emval will be registered multiple times for
+  // different type id's, but only a single type object is needed on the JS side
+  // for all of them. Store the type for reuse.
+  $EmValType__deps:  ['$createEmValType'],
+  $EmValType:  'createEmValType()',
```

Comment:
Why no just make this a JS object?   If there is some kind of problem evaluating the object at build time what we normally do use a mutli-line string.  e.g.:

```
$EmValType:  `{
  someObject: EmbeddedInString
}`,
```

Reply:
I tried that but it doesn't seem to work for the dependencies of `GenericWireTypeSize` and `simpleReadValueFromPointer` since they come up as undefined while building.

Reply:
Did you put the object literal in backtick quotes?

Reply:
Oh didn't see the backtick or know that was an option.